3rdparty/musl/CMakeLists.txt

```diff
@@ -9,54 +9,70 @@ set(PATCHES_DIR ${CMAKE_CURRENT_SOURCE_DIR}/patches)
-            CC=${CMAKE_C_COMPILER}
-            CXX=${CMAKE_CXX_COMPILER}
-    BUILD_COMMAND
-        ${CMAKE_COMMAND} -E chdir ${CMAKE_CURRENT_BINARY_DIR}/musl
-        make install-headers
-        COMMAND ${CMAKE_COMMAND} -E copy
-            ${OE_INCDIR}/openenclave/libc/endian.h
-            ${OE_INCDIR}/openenclave/libc/__endian.h
-        COMMAND ${CMAKE_COMMAND} -E copy
-            ${PATCHES_DIR}/endian.h
-            ${OE_INCDIR}/openenclave/libc/endian.h
-
-        # Append deprecations.h to all C header files.
-        COMMAND ${CMAKE_CURRENT_LIST_DIR}/append-deprecations
-            ${OE_INCDIR}/openenclave/libc
-
-        # Copy local deprecations.h to include/bits/deprecated.h.
-        COMMAND ${CMAKE_COMMAND} -E copy
-            ${CMAKE_CURRENT_LIST_DIR}/deprecations.h
-            ${OE_INCDIR}/openenclave/libc/bits/deprecations.h
-
-    BUILD_BYPRODUCTS
-        ${OE_INCDIR}/openenclave/libc ${CMAKE_CURRENT_BINARY_DIR}/musl
-    INSTALL_COMMAND "")
-
-# install-rule can be generated here, usage requirements defined in
-# <ROOT>/libc/.
-install(DIRECTORY
-    ${OE_INCDIR}/openenclave/libc
-    DESTINATION
-    ${CMAKE_INSTALL_INCLUDEDIR}/openenclave/3rdparty)
+ExternalProject_Add(musl_includes
+  DOWNLOAD_COMMAND
+    ${CMAKE_COMMAND} -E copy_directory
+    ${CMAKE_CURRENT_LIST_DIR}/musl
+    ${CMAKE_CURRENT_BINARY_DIR}/musl
+  PATCH_COMMAND
+    COMMAND ${CMAKE_COMMAND} -E copy
+      ${MUSL_DIR}/arch/x86_64/syscall_arch.h
+      ${MUSL_DIR}/arch/x86_64/__syscall_arch.h
+    COMMAND ${CMAKE_COMMAND} -E copy
+      ${PATCHES_DIR}/syscall_arch.h
+      ${MUSL_DIR}/arch/x86_64/syscall_arch.h
+    COMMAND ${CMAKE_COMMAND} -E copy
+      ${PATCHES_DIR}/pthread_arch.h
+      ${MUSL_DIR}/arch/x86_64/pthread_arch.h
+  CONFIGURE_COMMAND
+    ${CMAKE_COMMAND} -E chdir ${CMAKE_CURRENT_BINARY_DIR}/musl
+    ./configure
+      --includedir=${OE_INCDIR}/openenclave/libc
+      CFLAGS=${CFLAGS}
+      CC=${CMAKE_C_COMPILER}
+      CXX=${CMAKE_CXX_COMPILER}
+  BUILD_COMMAND
+    ${CMAKE_COMMAND} -E chdir ${CMAKE_CURRENT_BINARY_DIR}/musl
+    make install-headers
+    COMMAND ${CMAKE_COMMAND} -E copy
+      ${OE_INCDIR}/openenclave/libc/endian.h
+      ${OE_INCDIR}/openenclave/libc/__endian.h
+    COMMAND ${CMAKE_COMMAND} -E copy
+      ${PATCHES_DIR}/endian.h
+      ${OE_INCDIR}/openenclave/libc/endian.h
+
+    # Append deprecations.h to all C header files.
+    COMMAND ${CMAKE_CURRENT_LIST_DIR}/append-deprecations
+      ${OE_INCDIR}/openenclave/libc
+
+    # Copy local deprecations.h to include/bits/deprecated.h.
+    COMMAND ${CMAKE_COMMAND} -E copy
+      ${CMAKE_CURRENT_LIST_DIR}/deprecations.h
+      ${OE_INCDIR}/openenclave/libc/bits/deprecations.h
+
+  BUILD_BYPRODUCTS
+    ${OE_INCDIR}/openenclave/libc ${CMAKE_CURRENT_BINARY_DIR}/musl
+
+  INSTALL_COMMAND "")
+
+add_library(oelibc_includes INTERFACE)
+
+add_dependencies(oelibc_includes musl_includes)
+
+# NOTE: This is explicitly marked as a `SYSTEM` directory because it
```

Comment:
@anakrish This is the note I was talking about with respect to `SYSTEM` headers.